File: trunk/csv2code/src/shells/generate/lua/factions_power_data.lua
--
-- 官员定义表.csv
-- 官员编号 = id
-- 官员名称 = number
-- 官员权力 = deputyleader

local data = {
    [1] = {["id"] = 1, ["number"] = "首领", ["deputyleader"] = {1, 2, 3, 4, 5, 6, 7, 8, 9}},
    [2] = {["id"] = 2, ["number"] = "副首领", ["deputyleader"] = {1, 2, 3, 4, 5, 6, 7, 8}},
    [3] = {["id"] = 3, ["number"] = "外交官", ["deputyleader"] = {1, 2, 3, 4, 5, 6, 7}},
    [4] = {["id"] = 4, ["number"] = "内政官", ["deputyleader"] = {1, 2, 3, 4, 5}},
    [5] = {["id"] = 5, ["number"] = "精英", ["deputyleader"] = {1, 2, 3, 4, 5}},
    [6] = {["id"] = 6, ["number"] = "成员"}
}

return data

File: trunk/csv2code/src/shells/generate/java/FactionsPowerData.java
package com.gamejelly.gong2.config.data;

import com.gamejelly.gong2.config.data.base.LData;
import com.gamejelly.gong2.config.data.base.LMap;

/**
 * 官员定义表.csv
 * 官员编号 = id
 * 官员权力 = deputyleader
 */
public class FactionsPowerData extends LData {
    public static final LMap data = new LMap();

    static {
        init0();
        init1();
        data.setFrozen(true);
    }

    private static void init0() {
data.map(
    1, map("id", 1, "deputyleader", list(1, 2, 3, 4, 5, 6, 7, 8, 9)),
    2, map("id", 2, "deputyleader", list(1, 2, 3, 4, 5, 6, 7, 8)),
    3, map("id", 3, "deputyleader", list(1, 2, 3, 4, 5, 6, 7)),
    4, map("id", 4, "deputyleader", list(1, 2, 3, 4, 5)),
    5, map("id", 5, "deputyleader", list(1, 2, 3, 4, 5))
);
    }

    private static void init1() {
data.map(
    6, map("id", 6)
);
    }

}